diff --git a/src/plugin-main.cpp b/src/plugin-main.cpp
index 1faea82..7725e6b 100644
--- a/src/plugin-main.cpp
+++ b/src/plugin-main.cpp
@@ -412,19 +412,7 @@ const NDIlib_v6 *load_ndilib()
 	if (!temp_path.isEmpty()) {
 		locations << temp_path;
 	}
-#if defined(Q_OS_LINUX) || defined(Q_OS_MACOS)
-	// Linux, MacOS
-	// https://github.com/DistroAV/DistroAV/blob/master/lib/ndi/NDI%20SDK%20Documentation.pdf
-	// "6.1 LOCATING THE LIBRARY
-	// ... the redistributable on MacOS is installed within `/usr/local/lib` ..."
-	// Flatpak install will look for the NDI lib in /app/plugins/DistroAV/extra/lib
-	locations << "/usr/lib";
-	locations << "/usr/lib64";
-	locations << "/usr/local/lib";
-#if defined(Q_OS_LINUX)
-	locations << "/app/plugins/DistroAV/extra/lib";
-#endif
-#endif
+	locations << "@NDI@/lib";
 	auto lib_path = QString();
 #if defined(Q_OS_LINUX)
 	// Linux
